### PR TITLE
feat(public-cloud): gridscale changes on private network

### PR DESCRIPTION
**Pull request by sachinrameshn**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pci-localzone` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-12757 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/private-networks/private-networks.service.js [hunk 1]
```diff
@@ -45,6 +46,33 @@ export default class {
...
+                  id,
+                  allocatedIp,
+                  ...network,
+                });
+              },
+            );
+          }
+        });
+        return privateNetworks;
+      });
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/private-networks/private-networks.service.js:73]: **question:** Can we ask API team to add filter to distinguish between Local and Global private network. Also, please ask API team to add an indicator to determine the type of private network. We cannot rely on `vlan` as it is not explicit enough. In future, what if we have 3rd type of private network and how do you identify it?

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/private-networks/private-networks.service.js (hunk 1 above)
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/private-networks/private-networks.service.js:73]: Yup I will check with API team, if not I can check based on region.

File: packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.controller.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  $onInit() {
+    this.loadMessages();
+    this.isLoading = false;
+    this.criteria = getCriteria('id', this.networkId);
+    this.gatewaySubnetObj = null;
+    this.buildGatewaySubnetObj();
+  }
+
+  getSubnets(row) {
+    set(row, 'loading', true);
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.controller.js:26]:
```suggestion
    row.loading = true;
```

File: packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.controller.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      .all(
+        row.subnets.map(({ region, networkId }) =>
+          this.PciPrivateNetworks.getSubnets(this.projectId, region, networkId),
+        ),
+      )
+      .then((subnets) => {
+        set(
+          row,
+          'subnets',
+          row.subnets.map((subnet, index) => {
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.controller.js:37]:
```suggestion
          row.subnets = row.subnets.map((subnet, index) => {
```

File: packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.controller.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                allocatedIp: subnets[index].allocationPools
+                  .map((ipPool) => `${ipPool.start} - ${ipPool.end}`)
+                  .join(' ,'),
+                gatewayName: this.gatewaySubnetObj[subnet.networkId],
+              };
+            }
+            return subnet;
+          }),
+        );
+        set(row, 'loading', false);
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.controller.js:51]:
```suggestion
        row.loading = false;
```

File: packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.controller.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        return row;
+      })
+      .catch((error) => {
+        set(row, 'loading', false);
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.controller.js:55]:
```suggestion
        row.loading = false;
```

File: packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.routing.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            .injector()
+            .getAsync('vrack')
+            .then((vrack) =>
+              isEmpty(vrack)
+                ? { state: 'pci.projects.project.privateNetwork.vrack' }
+                : false,
+            );
+        }),
+    resolve: {
+      deleteSubnet: /* @ngInject */ (
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.routing.js:33]:
```suggestion
      goToDeleteSubnet: /* @ngInject */ (
```

File: packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.component.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import controller from './global-regions.controller';
+import template from './global-regions.html';
+
+export default {
+  controller,
+  template,
+  bindings: {
+    createNetwork: '<',
+    deleteSubnet: '<',
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.component.js:9]:
```suggestion
    goToDeleteSubnet: '<',
```

File: packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.html [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+                                >
+                                    <i
+                                        class="oui-icon oui-icon-gear"
+                                        aria-hidden="true"
+                                    ></i>
+                                </button>
+                                <button
+                                    class="oui-button oui-button_ghost"
+                                    data-ng-if="!$row.loading"
+                                    data-ng-click="$ctrl.deleteSubnet(subnetDetail.networkId, subnetDetail.region)"
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/global-regions/global-regions.html:153]:
```suggestion
                                    data-ng-click="$ctrl.goToDeleteSubnet(subnetDetail.networkId, subnetDetail.region)"
```

File: packages/manager/modules/pci/src/projects/project/private-networks/local-zone/local-zone.controller.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  constructor(CucCloudMessage) {
+    this.CucCloudMessage = CucCloudMessage;
+    this.PRIVATE_NETWORK_LIST = PRIVATE_NETWORK_LIST;
+  }
+
+  $onInit() {
+    this.loadMessages();
+    this.isLoading = false;
+  }
+
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/local-zone/local-zone.controller.js:15]:
```suggestion
$onDestroy() {
    this.cucCloudMessage.unSubscribe(CONTAINER_NAME);
}
```

File: packages/manager/modules/pci/src/projects/project/private-networks/local-zone/local-zone.html [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        data-property="allocatedIp"
+        data-searchable
+        data-type="string"
+        data-filterable
+    ></oui-datagrid-column>
+    <oui-datagrid-column>
+        <button
+            class="oui-button oui-button_ghost"
+            data-ng-if="!$row.loading"
+            data-ng-click="$ctrl.deleteSubnet($row.id, $row.region)"
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/local-zone/local-zone.html:75]:
```suggestion
            data-ng-click="$ctrl.goToDeleteSubnet($row.id, $row.region)"
```

File: packages/manager/modules/pci/src/projects/project/private-networks/local-zone/local-zone.routing.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('pci.projects.project.privateNetwork.localZone', {
+    url: '/localZone',
+    views: {
+      containersView: 'pciPrivateNetworksLocalZone',
+    },
+    resolve: {
+      deleteSubnet: /* @ngInject */ (
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/local-zone/local-zone.routing.js:8]:
```suggestion
      goToDeleteSubnet: /* @ngInject */ (
```

File: packages/manager/modules/pci/src/projects/project/private-networks/local-zone/local-zone.component.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+import controller from './local-zone.controller';
+import template from './local-zone.html';
+
+export default {
+  controller,
+  template,
+  bindings: {
+    createNetwork: '<',
+    deleteSubnet: '<',
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/private-networks/local-zone/local-zone.component.js:9]:
```suggestion
    goToDeleteSubnet: '<',
```

**Review by qpavy — APPROVED** [review]

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/private-networks/private-networks.html [hunk 13]
```diff
@@ -17,8 +17,25 @@
                 ></span>
             </oui-guide-menu-item>
         </oui-guide-menu>
-    </oui-header>
 
+        <oui-header-tabs data-ng-if="$ctrl.privateNetworks.length !== 0">
+            <oui-header-tabs-item
+                data-href="{{:: $ctrl.globalRegionsLink }}"
+                data-active="!$ctrl.isLocalZoneTabActive()"
```
[the remark this answers is not in the document]
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/private-networks/private-networks.html]: Done Anoop

File: packages/manager/modules/pci/src/projects/project/private-networks/private-networks.html [hunk 14]
```diff
@@ -17,8 +17,25 @@
                 ></span>
             </oui-guide-menu-item>
         </oui-guide-menu>
-    </oui-header>
 
+        <oui-header-tabs data-ng-if="$ctrl.privateNetworks.length !== 0">
+            <oui-header-tabs-item
+                data-href="{{:: $ctrl.globalRegionsLink }}"
+                data-active="$ctrl.currentActiveLink() === $ctrl.globalRegionsLin"
```
[the remark this answers is not in the document]
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/private-networks/private-networks.html:24]: my bad Good catch !!

**Review by anooparveti — APPROVED** [review]

**Pull request merged**